frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -339,9 +350,10 @@ const DeviceUserPage = ({
               </Tabs>
             </TabsWrapper>
             {showInfoModal && <InfoModal onCancel={toggleInfoModal} />}
-            {showMdmModal && (
-              <ManualEnrollMdmModal
-                onCancel={toggleTurnOnMdmModal}
+            {showEnrollMdmModal && (
+              <EnrollMdmModal
+                mdmEnrollmentType={mdmEnrollmentType as "manual" | "auto"}
```

Comment:
can we just leave this as `mdmEnrollmentType` and change the props to `mdmEnrollmentType?: "manual" | "auto";`

?

Reply:
Yea, this was a prickly little area – because I'm using mdmEnrollmentType's truthiness to determine whether to show the modal, it is potentially undefined. However at runtime, it will never actually be undefined, so I wanted the modals's prop interface to clearly reflect that – if undefined is getting passed into EnrollMdmModal, there is a problem, that should not happen. But, on the DeviceUserPage level, I am using mdmEnrollmentType's truthiness to determine whether to show the modal at all, so it CAN be undefined (falsey), but in that situation the modal will never be rendered. So I'm basically saying "At DeviceUserPage scope, this var might be undefined, but once it's being passed into EnrollMdmModal, it will never be undefined, only "manual" | "auto". 

If you see a more graceful solution that would be great, but this is where I arrived after considering a few options.

Reply:
Ya, good frontend conversation. We want to be consistent with our patterns, would love to hear from @ghernandez345 what is generally preferred and why.

Reply:
for this case, I think it makes more sense to ensure a value in the condition above. That way we are adhering to `EnrollMdmModal` prop interface and not overriding the compiler checks. 

```tsx
{showEnrollMdmModal && mdmEnrollmentType (
  <EnrollMdmModal mdmEnrollmentType={mdmEnrollmentType} >
```
how it's currently written may not seem like a big deal but we are overriding TS compiling checks and if in the future `mdmEntollmentType` can be another value then our compiler would not show us that our typing is incorrect. Overriding the compiling rules with `as` is sometimes valid to do (tbh it's usually to save time) but as the change here is quite quick, id drop it.